api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -68,4 +69,8 @@ message UdpProxyConfig {
   // load balancing algorithms will select a host randomly. Currently the number of hash policies is
   // limited to 1.
   repeated HashPolicy hash_policies = 5 [(validate.rules).repeated = {max_items: 1}];
+
+  // The maximum size of received UDP packets. Using a larger size will cause Envoy to allocate
+  // more memory per upstream. If not set defaults to 1500 bytes.
+  google.protobuf.UInt64Value max_packet_size = 6;
```

Comment:
Should this one be on the upstream/cluster instead of on the proxy? If it needs to be here, I think it should be named `max_upstream_packet_size` or similar. 

Also, should possibly contain `rx` or `received` in the name. Setting this does not change the size of transmitted packets if the two sides have different limits.

Reply:
I think we have some options here as I'm hesitant to put this directly on the cluster as we are trying to remove extension specific configuration from the cluster in general.
1) We could possibly add extension specific cluster metadata and store this value there.
2) We can leave this here as it's somewhat specific to UDP proxy which originates these datagram "connections."
3) We could potentially have this configuration be route/cluster specific within the UDP proxy config, especially if we eventually support routing rules there.

In thinking about, in general I prefer (2) or (3), and since we don't yet support routing I mildly prefer to keep it simple and stick with (2). WDYT?

Regarding naming, I agree with adding rx/received, though WDYT about changing both params to max_rx_packet_size (listener and here) so that we keep the name consistent but it's more clear which direction we are talking about? Otherwise I would be inclined to change the other one to max_downstream_rx_packet_size and this one to max_upstream_rx_packet_size. Which do you prefer?

Reply:
I'm fine with 2 or 3.

I think naming max_downstream_rx_packet_size and max_upstream_upstream_rx_packet_size is better so it's clear what it applies to. On the listener it's clear; on the proxy it seems very unclear in my mind (especially if there's the possibility that there are upstream filters between the proxy and the upstream connection). I think the idea of upstream filters is why it makes less sense to me to be on the proxy, but I understand not wanting to heap a bunch of stuff into the cluster.

Reply:
I thought about it more and I think for clarity I would like to switch to:
1) max_downstream_rx_datagram_size on the listener
2) max_upstream_rx_datagram_size on the UDP proxy

LMK what you think about this.

Reply:
Sorry swapped comments with you. OK sounds like we are on the same page, as long as you are fine switching from packet -> datagram.

For UDP we don't support upstream filters at all right now, it's all terminated directly in the UDP proxy filter. I agree in the future it might get more complex but I guess my feeling is to keep it simple for now.